### ISSUE TITLE
Fine-tuning camemBERT for similarities between sentences

### ISSUE THREAD

**Issue by QuentinSpalla** [UKPLab/sentence-transformers#110]
Hi,

My goal here is to do clustering on sentences. For this purpose, I chose to use similarities between sentence embedding for all my sentences. Unfortunately, camemBERT is not great for that task and SBERT article seems really helpful.

I managed to fine tune camemBERT on an NLI (thanks to that [function](https://github.com/getalp/Flaubert/blob/master/flue/get-data-xnli.sh) and [this one](https://github.com/getalp/Flaubert/blob/master/flue/prepare-data-xnli.sh)).
I trained the model on Google Cloud Platform with an NVIDIA Tesla P100 for a few hours and 6 epochs by copying only the [training_nli_bert.py](https://github.com/UKPLab/sentence-transformers/blob/master/examples/training_nli_bert.py) and changing just a few things:

```
"""
The system trains BERT on the SNLI + MultiNLI (AllNLI) dataset
with softmax loss function. At every 1000 training steps, the model is evaluated on the
STS benchmark dataset
"""
from torch.utils.data import DataLoader
import math
from sentence_transformers import models, losses
from sentence_transformers import SentencesDataset, LoggingHandler, SentenceTransformer
from sentence_transformers.evaluation import LabelAccuracyEvaluator
from sentence_transformers.readers import *
import logging
from datetime import datetime

#### Just some code to print debug information to stdout
logging.basicConfig(format='%(asctime)s - %(message)s',
                    datefmt='%Y-%m-%d %H:%M:%S',
                    level=logging.INFO,
                    handlers=[LoggingHandler()])
#### /print debug information to stdout

# Read the dataset
model_name = 'camembert-base'
batch_size = 16
nli_reader = NLIDataReader('/model/flaubert/data/processed')
# sts_reader = STSDataReader('datasets/stsbenchmark')
train_num_labels = nli_reader.get_num_labels()
model_save_path = '/model/camembert/sentence_transformer/output/training_nli_'+model_name+'-'+datetime.now().strftime("%Y-%m-%d_%H-%M-%S")



# Use BERT for mapping tokens to embeddings
word_embedding_model = models.CamemBERT(model_name)
print("END TOKENIZING")

# Apply mean pooling to get one fixed sized sentence vector
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension(),
                               pooling_mode_mean_tokens=True,
                               pooling_mode_cls_token=False,
                               pooling_mode_max_tokens=False)

model = SentenceTransformer(modules=[word_embedding_model, pooling_model])


# Convert the dataset to a DataLoader ready for training
logging.info("Read AllNLI train dataset")
print("reading data")
train_data = SentencesDataset(nli_reader.get_examples('train'), model=model)
train_dataloader = DataLoader(train_data, shuffle=True, batch_size=batch_size)
train_loss = losses.SoftmaxLoss(model=model, sentence_embedding_dimension=model.get_sentence_embedding_dimension(), num_labels=train_num_labels)
print("END reading data")


logging.info("Read NLI dev dataset")
dev_data = SentencesDataset(examples=nli_reader.get_examples('valid'), model=model)
dev_dataloader = DataLoader(dev_data, shuffle=False, batch_size=batch_size)
evaluator = LabelAccuracyEvaluator(dev_dataloader, softmax_model=train_loss)

# Configure the training
num_epochs = 6

warmup_steps = math.ceil(len(train_dataloader) * num_epochs * 0.1) #10% of train data for warm-up
logging.info("Warmup-steps: {}".format(warmup_steps))



# Train the model
model.fit(train_objectives=[(train_dataloader, train_loss)],
          evaluator=evaluator,
          epochs=num_epochs,
          evaluation_steps=2000,
          warmup_steps=warmup_steps,
          output_path=model_save_path
          )



##############################################################################
#
# Load the stored model and evaluate its performance on STS benchmark dataset
#
##############################################################################

model = SentenceTransformer(model_save_path)
test_data = SentencesDataset(examples=nli_reader.get_examples("test"), model=model)
test_dataloader = DataLoader(test_data, shuffle=False, batch_size=batch_size)
evaluator = LabelAccuracyEvaluator(test_dataloader, softmax_model=train_loss)

model.evaluate(evaluator)

```

After my training, the accuracy for the evaluator LabelAccuracyEvaluator is 0.7 which is not high...
Does anyone has an advice concerning the hyperparameters of the training that could help me increase the accuracy ?

Thanks in advance

**Comment by nreimers** [comment]
Hi @QuentinSpalla ,
CamamBERT is designed for French, hence you should fine-tune it on a French dataset (if you want to use it for french sentences).

Note, I'm currently release a pre-trained multilingual model, that also works with French (any many other languages):
```
model = SentenceTransformer('distiluse-base-multilingual-cased')
```

The documentation on this model will be added soon.

Best
Nils Reimers

**Comment by QuentinSpalla** [comment]
Hi,

> CamamBERT is designed for French, hence you should fine-tune it on a French dataset (if you want to use it for french sentences).

@nreimers: Yes i did my fine-tuning on a french dataset (links in my previous comment). Could you tell me what is your parameters configuration for fine-tuning on an nli dataset please (no matter which language).

> Note, I'm currently release a pre-trained multilingual model, that also works with French (any many other languages)

Is it ready to use ? thank you very much it is very useful

Thanks

**Comment by nreimers** [comment]
Hi @QuentinSpalla 
You can see the training on the NLI dataset here (for the English version):
https://github.com/UKPLab/sentence-transformers/blob/master/examples/training_nli_bert.py

Here the details on the multi-lingual model:
https://github.com/UKPLab/sentence-transformers/blob/master/docs/pretrained-models/multilingual-models.md

It is ready to use. I haven't tested the quality on STS for French (any good datasets available)? But for German, Spanish, Arabic and Turkish it yields good results. It should also work well for French.

Best
Nils Reimers

**Comment by QuentinSpalla** [comment]
Hi @nreimers,

Thanks for all.
